### ISSUE TITLE
Missing common overridable/callable method summaries in docs

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4815]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)| @jyknight reported|
|-|-|
|Trac ID|trac#222|
|Type|enhancement|
|Created|2003-08-22 02:43:24Z|

<details><summary>Searchable metadata</summary>

```
trac-id__222 222
type__enhancement enhancement
reporter__jknight jknight
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1061520204000000 1061520204000000
changetime__1468742627053789 1468742627053789
version__ 
owner__ 
cc__spiv cc__jknight cc__hypatia. cc__exarkun
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)|@jyknight commented|
|-|-|

```
#!html
<pre>
There is no section "Things that can happen to your Protocol" and 
"Things you can do from your Protocol" in the Server and Client sections, 
to correspond to the sections (3.6.[45] in ProcessProtocol. I assumed 
when reading the earlier sections that the omission was deliberate, and 
that I should read the API docs to get that info. But, if you're going to 
have it in one section, put it in all of them.

It is somewhat useful to see it (since I haven't even started 
looking at the API docs), so I'd suggest adding it to the other sections 
instead of removing it from the ProcessProtocol section. :)

</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv commented|
|-|-|

```
#!html
<pre>
Probably the docs should refer you to the API docs for the
IProtocol interface.

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @jyknight|
|-|-|

Is this the sort of thing you had in mind?  If so, should something similar be added to other documents?  Which?

```
Index: clients.xhtml
===================================================================
--- clients.xhtml       (revision 18799)
+++ clients.xhtml       (working copy)
@@ -261,6 +261,26 @@
     self.factory. It can then access attributes of the factory in its logic.
     In the case of LogBot, it opens the file and connects to the channel
     stored in the factory.</p>
-    
+
+    <h2>Further Reading</h2>
+
+    <p>The <code class="API"
+    base="twisted.internet.protocol">Protocol</code> class used throughout
+    this document is a base implementation of <code class="API"
+    base="twisted.internet.interfaces">IProtocol</code> used in most Twisted
+    applications for convenience.  To learn about the complete
+    <code>IProtocol</code> interface, see the API documentation for <code
+    class="API" base ="twisted.internet.interfaces">IProtocol</code>.</p>
+
+    <p>The <code>transport</code> attribute used in some examples in this
+    document provides the <code class="API"
+    base="twisted.internet.interfaces">ITCPTransport</code> interface.  To
+    learn about the complete interface, see the API documentation for <code
+    class="API" base="twisted.internet.interfaces">ITCPTransport</code>.</p>
+
+    <p>Interface classes are a way of specifying what methods and attributes
+    an object has and how they behave.  See the <a href="components.xhtml">
+    Components: Interfaces and Adapters</a> document.</p>
+
   </body>
 </html>
```

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_4"></a>|@jerub set owner to spiv|
|-|-|

In the absence from a response from jknight, I approve the above documentation change. Please apply the patch.

Merry Christmas.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [20652]) Apply protocol documentation patch from [#4815](https://github.com/twisted/twisted/issues/4815)

Author: exarkun
Reviewer: Jerub
Fixes [#4815](https://github.com/twisted/twisted/issues/4815)

Add some text to the end of the client howto explaining some
conventions used in the document and where further information
can be found.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_7"></a>|GitHub commented|
|-|-|

In changeset 4df5aae23fb3945d6555d47280e09c85ae6e7fa2
```
#!CommitTicketReference repository="" revision="4df5aae23fb3945d6555d47280e09c85ae6e7fa2"
Merge pull request #222 from markrwilliams/pypy-persisted-styles-8504

Author: markrwilliams
Reviewer: rodrigc
Fixes: #8504

twisted.persisted.styles._pickleFunction explicitly raises a PicklingError when given a lambda.
```
